numpy/ma/core.py

```diff
@@ -3184,13 +3180,17 @@ def __setitem__(self, indx, value):
 
         """
         # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
+        self._oldsharedmask = getattr(self, "_oldsharedmask", False)
+        self._oldsharedmask = self._oldsharedmask or self._sharedmask
```

Comment:
Right, it was what I figured out to handle this case.

    import numpy as np

    a = np.ma.masked_array(np.zeros((2,3)), mask=np.zeros((2,3)))
    b = a[0:1]
    
    b[0] = np.ma.masked      # raises the warning (mask gets copied)
    b[0,1] = np.ma.masked    # doesn't raise a warning